packages/at_root_server/lib/src/server/root_server_bootstrapper.dart

```diff
@@ -7,36 +7,47 @@ import 'package:at_persistence_root_server/at_persistence_root_server.dart';
 import 'package:at_utils/at_logger.dart';
 
 class RootServerBootStrapper {
-  static final bool? useSSL = AtRootConfig.useSSL;
-  var arguments;
-  var logger = AtSignLogger('RootServerBootStrapper');
+  final logger = AtSignLogger('RootServerBootStrapper');
 
-  RootServerBootStrapper(this.arguments);
+  late final AtRootServerContext _rootContext;
 
-  Future<void> run() async {
-    try {
-      var results = CommandLineParser().getParserResults(arguments);
-      var rootContext = AtRootServerContext();
-      rootContext.port = AtRootConfig.rootServerPort;
-      rootContext.redisServerHost = results['redis_host'];
-      rootContext.redisServerPort = int.parse(results['redis_port']);
-      rootContext.redisAuth = results['redis_auth'];
-      if (useSSL!) {
-        rootContext.securityContext = AtSecurityContextImpl();
-      }
-      // Initialize the ConnectionManager for the key store with redisHost and port
-      var redisManager = RedisConnectionManager.getInstance();
-      await redisManager.init(rootContext.redisServerHost!,
-          rootContext.redisServerPort!, rootContext.redisAuth!);
-      var keyStoreManager = KeystoreManagerImpl();
-      var result = await (keyStoreManager.getKeyStore().get('ping'));
-      logger.info(result);
-      assert(result != null && 'pong'.compareTo(result) == 0);
-      var serverInstance = RootServerImpl();
-      serverInstance.setServerContext(rootContext);
-      serverInstance.start();
-    } on Exception {
-      rethrow;
+  AtRootServerContext get rootContext => _rootContext;
+
+  RootServerBootStrapper(List<String> args) {
+    _rootContext = createAtRootServerContext(args);
+  }
+
+  static AtRootServerContext createAtRootServerContext(List<String> args) {
+    final AtRootServerContext c = AtRootServerContext();
+
+    c.port = AtRootConfig.rootServerPort;
+    c.httpsPort = AtRootConfig.httpsPort;
+    c.httpsEnabled = AtRootConfig.httpsEnabled;
+
+    var results = CommandLineParser().getParserResults(args);
+    c.redisServerHost = results['redis_host'];
+    c.redisServerPort = int.parse(results['redis_port']);
+    c.redisAuth = results['redis_auth'];
+
+    c.useSSL = AtRootConfig.useSSL;
+    if (c.useSSL!) {
+      c.securityContext = AtRootSecurityContextImpl();
     }
+
+    return c;
+  }
+
+  Future<void> run() async {
+    // Initialize the ConnectionManager for the key store with redisHost and port
+    var redisManager = RedisConnectionManager.getInstance();
+    await redisManager.init(rootContext.redisServerHost!,
+        rootContext.redisServerPort!, rootContext.redisAuth!);
+    var keyStoreManager = KeystoreManagerImpl();
+    var result = await (keyStoreManager.getKeyStore().get('ping'));
+    logger.info(result);
+    assert(result != null && 'pong'.compareTo(result) == 0);
+    var serverInstance = RootServerImpl();
+    serverInstance.setServerContext(rootContext);
+    serverInstance.start();
```

Comment:
Moved stuff around here so we could unit test against different environments & yaml configs